packages/loopback/test/acceptance/context-bindings/class-level-bindings.feature.md

```diff
@@ -0,0 +1,33 @@
+# Feature: Context bindings - injecting dependencies of classes
+
+- In order to manage my dependencies
+- As a developer
+- I want to setup bindings for my classes
+- So that class dependencies are injected by the IoC framework
+
+## Scenario: Inject constructor arguments
+
+ - Given a context
+ - Given class `InfoController` with a constructor
+    accepting a single argument `appName: string`
+ - Given `InfoController` ctor argument is decorated
+     with `@inject('application.name')`
+ - Given a static binding named `application.name` with value `CodeHub`
+ - Given a class binding named `controllers.info` bound to class `InfoController`
+ - When I resolve the binding for `controllers.info`
+ - Then I get a new instance of `InfoController`
+ - And the instance was created with `appName` set to `CodeHub`
+
+ ```ts
+ const cxt = new Context();
+ ctx.bind('application.name').to('CodeHub');
+
+ class InfoController {
+   constructor(@inject('application.name') public appName: string) {
+   }
+ }
+ ctx.bindClass('controllers.info', InfoController);
```

Comment:
What about the other styles?

- `ctx.bind('controllers.info').to(InfoController);`
- `ctx.bind('controllers.info').toClass(InfoController);`